src/core/cluster.rs

```diff
@@ -127,7 +214,7 @@ impl<'a, T: Number, U: Number, D: Dataset<T, U>> Cluster<'a, T, U, D> {
     /// * `indices`: The indices of instances from the `dataset` that are
     /// contained in the `Cluster`.
     /// * `name`: `BitVec` name for the `Cluster`.
-    pub fn new(data: &'a D, indices: Vec<usize>, history: BitVec, seed: Option<u64>) -> Self {
+    pub fn new(data: &D, indices: Vec<usize>, history: BitVec, seed: Option<u64>) -> Self {
```

Comment:
Change the type of `indices` to `&[size]` and call `.to_vec()` when instantiating the `Cluster`